### PR TITLE
update mdev devices when the layout on the host changes

### PR DESCRIPTION
**Pull request by vladikr**
**What this PR does / why we need it**:
So far an update of the mediated devices on the nodes could have been triggered by 
 - an update of the KubeVirt CR
 - a change of node labels that matched the `nodeMediatedDevicesTypes` node selector.
Mediated devices would then be created based on the availability of requested types in a relevant sysfs path.
However, these types would appear in sysfs when a relevant vendor driver is installed.

In some cases, when the driver is being installed after an update of the KubeVirt CR, mediated devices would not be created by the virt-handler.

This PR ake the virt-handler to actively scan the relevant sysfs to discover changes in the availability of the mdev types.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # RHBZ#2046298

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/device-manager/mediated_devices_types.go [hunk 1]
```diff
@@ -72,6 +80,7 @@ func (m *MDEVTypesManager) updateMDEVTypesConfiguration(desiredTypesList []strin
 		}
 		// store the configured list of types
 		m.configuredMdevTypes = desiredTypesBytes
+		m.discoveredMdevTypesCount = len(files)
```
Comment (rmohr) [inline pkg/virt-handler/device-manager/mediated_devices_types.go:83]: I am not entirely sure if just taking the count is good enough. Would it make sense to just always create the updated list and then just compare if something changed?

Comment (vladikr) [reply pkg/virt-handler/device-manager/mediated_devices_types.go:83]: Yes, I went back and forth on this.
The issue is that `discoverConfigurableMDEVTypes` may be dependent on the results of `removeUndesiredMDEVs`
But,  `removeUndesiredMDEVs` should run only if there was a change in `desiredTypesBytes`
That's why I thought that counting the types would be an easy workaround, but you're right. It's not enough.

On the other hand, I think just keeping a list of PCI cards that provide mdevs, should be enough.
I'll post an update.

While looking into this again, I've discovered that we're not keeping track of cards that are already configured. That way updateMDEVTypesConfiguration may not take it into account correctly.
I'll update this as well now.

**Pull request closed without merging**